File: BOJ/1449.cpp
// 1449 수리공 항승
#include <iostream>
#include <algorithm>

using namespace std;

int solution(int n, int l, int arr[]) {
    int answer = 0;

    sort(arr, arr + n);
    int cmp = arr[0];
    for (int i = 1; i < n; ++i) {
        if (l <= arr[i] - cmp) {
            answer++;
            cmp = arr[i];
        }
    }

    ++answer;
    return answer;
}

int main(void)
{
    int n, l;
    int arr[1001];

    cin >> n >> l;
    for (int i = 0; i < n; ++i) {
        cin >> arr[i];
    }

    int ret = solution(n, l, arr);
    cout << ret;
    return 0;
}

File: BOJ/1449.py
# 1449 수리공 항승
from sys import stdin 
input = lambda : stdin.readline()

n, l = map(int, input().split())
pipe = list(map(int, input().split()))

pipe.sort() # 오름차순 정렬
cmp = pipe[0] # 기준
answer = 1

for cur in pipe[1:]:
    # 현재 테이프로 붙일 수 있는 경우
    if l <= cur - cmp:
        answer += 1
        cmp = cur

print(answer)